cmd/kubeadm/app/cmd/join.go

```diff
@@ -499,9 +499,9 @@ func fetchInitConfigurationFromJoinConfiguration(cfg *kubeadmapi.JoinConfigurati
 	// Create the final KubeConfig file with the cluster name discovered after fetching the cluster configuration
 	clusterinfo := kubeconfigutil.GetClusterFromKubeConfig(tlsBootstrapCfg)
 	tlsBootstrapCfg.Clusters = map[string]*clientcmdapi.Cluster{
-		initConfiguration.ClusterName: clusterinfo,
+		initConfiguration.Name: clusterinfo,
```

Comment:
Uhmm, is it correct?
As far as I understood, the name in initConfiguration has no real meaning; Should ClusterConfiguration.name be used instead?

Reply:
`ClusterConfiguration.ObjectMeta.Name` is being used there, although through `InitConfiguration`.
In the internal types ClusterConfiguration is embedded in InitConfiguration and its fields can be used directly from there. This is a backlog technical dept that is a thorn in my eyes, but I don't have the time to address it.

Reply:
This is really confusing, but I got your point. Let's just open an issue to track this